packages/contracts-bedrock/snapshots/state-diff/Kontrol-Deploy.json

```diff
@@ -1,3 +1,3 @@
684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d61035473ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff161480610509575033155b1561028e5761028b8161060c565b60006105417fb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d61035490565b73ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff161480610578575033155b156104a557507fb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d61035490565b7f360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc81815560405173ffffffffffffffffffffffffffffffffffffffff8316907fbc7cd75a20ee27fd9adebab32041f755214dbc6bffa90cc0225b39da2e5c2d3b90600090a25050565b60006106367fb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d61035490565b7fb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d61038381556040805173ffffffffffffffffffffffffffffffffffffffff80851682528616602082015292935090917f7e644d79422f17c01e4894b5f4f588d331ebfa28653d42ae832dc59e38c9798f910160405180910390a1505050565b803573ffffffffffffffffffffffffffffffffffffffff811681146106d857600080fd5b919050565b6000602082840312156106ef57600080fd5b610412826106b4565b60008060006040848603121561070d57600080fd5b610716846106b4565b9250602084013567ffffffffffffffff8082111561073357600080fd5b818601915086601f83011261074757600080fd5b81358181111561075657600080fd5b87602082850101111561076857600080fd5b6020830194508093505050509250925092565b600060208083528351808285015260005b818110156107a85785810183015185820160400152820161078c565b818111156107ba576000604083870101525b50601f017fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe016929092016040019392505050565b818382376000910190815291905056fea164736f6c634300080f000ab53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103000000000000000000000000db8cff278adccf9e9b5da745b44e754fc4ee3c76","deployedCode":"0x60806040526004361061005e5760003560e01c80635c60da1b116100435780635c60da1b146100be5780638f283970146100f8578063f851a440146101185761006d565b80633659cfe6146100755780634f1ef286146100955761006d565b3661006d5761006b61012d565b005b61006b61012d565b34801561008157600080fd5b5061006b6100903660046106dd565b610224565b6100a86100a33660046106f8565b610296565b6040516100b5919061077b565b60405180910390f35b3480156100ca57600080fd5b506100d3610419565b60405173ffffffffffffffffffffffffffffffffffffffff90911681526020016100b5565b34801561010457600080fd5b5061006b6101133660046106dd565b6104b0565b34801561012457600080fd5b506100d3610517565b60006101577f360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc5490565b905073ffffffffffffffffffffffffffffffffffffffff8116610201576040517f08c379a000000000000000000000000000000000000000000000000000000000815260206004820152602560248201527f50726f78793a20696d706c656d656e746174696f6e206e6f7420696e6974696160448201527f6c697a656400000000000000000000000000000000000000000000000000000060648201526084015b60405180910390fd5b3660008037600080366000845af43d6000803e8061021e573d6000fd5b503d6000f35b7fb53127684a568b3173ae13b9f8a6016e243e63b6e8ee
```

Comment:
We should prettify this JSON file to make the diff easier to review in the future, but I need to check if we have a JSON formatter that's typically used in this repo. From a quick look I didn't see a script for it in the `package.json`

I'll manually format/review this diff on Monday.

Reply:
Diff LGTM, rationale below:

Attached are sceenshots are the 4 changes. The only diff is the slot numbers in contract `0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f`
- Slot 0x35 (53) became 0x37 (55)
- Slot 0x36 (54) became 0x38 (56)

Address 0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f is the default test contract (I.e. the first one deployed) and it seems this is the `DeployConfig` contract based on the below:

Two slots were added to the `DeployConfig` contract recently in https://github.com/ethereum-optimism/optimism/pull/9039/files#diff-fcedbe6005fe092aab124a65c9eb9a2c5824c1cd83461b144daea68b6c8a4d2bR56-R57

- Selector `0x38c27159` is the `requiredProtocolVersion` method on the `DeployConfig` contract. Sanity checking by running ``solc `forge re` scripts/DeployConfig.s.sol --storage-layout`` confirms that the `requiredProtocolVersion is` is now in slot 56)
- Selector `0x67cef446` is the `recommendedProtocolVersion` method on the `DeployConfig` contract. Sanity checking by runningthe same command confirms that the `recommendedProtocolVersion` is now in slot 56)

<img width="788" alt="image" src="https://github.com/ethereum-optimism/optimism/assets/17163988/c3bcc71c-5148-420a-8f1f-878c662bb88f">
<img width="789" alt="image" src="https://github.com/ethereum-optimism/optimism/assets/17163988/bbba8b2a-bd94-40da-b37d-ec2e2dd5c1df">
<img width="759" alt="image" src="https://github.com/ethereum-optimism/optimism/assets/17163988/1bc2aa91-24a9-4999-b241-411927339608">
<img width="786" alt="image" src="https://github.com/ethereum-optimism/optimism/assets/17163988/31e8d76a-ff90-47db-aa4d-efce11f61fe8">

Reply:
Awesome! Thanks for looking into this!